folder_paths.py

```diff
@@ -68,6 +68,41 @@ def get_directory_by_type(type_name):
     return None
 
 
+# determine base_dir rely on annotation if name is 'filename.ext [annotation]' format
+# otherwise use default_path as base_dir
+def get_annotated_filepath(name, default_dir=None):
+    if name.endswith("[output]"):
+        base_dir = get_output_directory()
+        name = name[:-9]
+    elif name.endswith("[input]"):
+        base_dir = get_input_directory()
+        name = name[:-8]
+    elif name.endswith("[temp]"):
+        base_dir = get_temp_directory()
+        name = name[:-7]
```

Comment:
Can you move this code into another function so that there's less code duplication with your exists_annotated_filepath function?

Reply:
refactoring done.